File: gsb/gsb-bd/src/main/java/com/gongsibao/supplier/web/SupplierListPart.java
package com.gongsibao.supplier.web;

import org.netsharp.communication.ServiceFactory;
import org.netsharp.panda.commerce.ListPart;

import com.gongsibao.supplier.base.ISupplierService;

public class SupplierListPart extends ListPart{

	ISupplierService supplierService = ServiceFactory.create(ISupplierService.class);
	
	/**   
	 * @Title: open   
	 * @Description: TODO(开户)   
	 * @param: @param supplierId
	 * @param: @return      
	 * @return: Boolean      
	 * @throws   
	 */
	public Boolean openAccount(Integer supplierId){
		
		return supplierService.openAccount(supplierId);
	}
	
	/**   
	 * @Title: close   
	 * @Description: TODO(销户)   
	 * @param: @param supplierId
	 * @param: @return      
	 * @return: Boolean      
	 * @throws   
	 */
	public Boolean closeAccount(Integer supplierId){
		
		return supplierService.closeAccount(supplierId);
	}
}

File: gsb/classes/artifacts/gsb-web/exploded/gsb-web-2.3.9-SNAPSHOT.war/gsb/supplier/js/supplier-list-part.js
System.Declare("com.gongsibao.supplier.web");
com.gongsibao.supplier.web.SupplierListPart = org.netsharp.panda.commerce.ListPart.Extends( {

    ctor: function () {
        this.base();
    },

	onAdding : function() {

    	var relationItem = this.relationItem;
    	if(relationItem == null){
    		
    		IMessageBox.info("请选择服务商分类");
    		return false;
    	}
    	
    	if(relationItem.children.length>0){
    		
    		IMessageBox.info("请选择末节点");
    		return false;
    	}
    	return true;
    },
    openAccount:function(){//开户
    	
		var count = this.getSelectionCount();
		if (count <= 0) {

			IMessageBox.warning("请选择要操作的记录！");
			return;
		}

		var ids = this.getSelectionIds();
		var me = this;
		IMessageBox.confirm("确认要提交吗？", function(istrue) {

			if (istrue) {
				me.invokeService("openAccount", [ids], function(data) {

					me.reload();
					IMessageBox.toast("开户成功！");
				});
			}
		});
    },
    closeAccount:function(){//销户
    	
		var count = this.getSelectionCount();
		if (count <= 0) {

			IMessageBox.warning("请选择要操作的记录！");
			return;
		}

		var ids = this.getSelectionIds();
		var me = this;
		IMessageBox.confirm("确认要提交吗？", function(istrue) {

			if (istrue) {
				me.invokeService("closeAccount", [ids], function(data) {

					me.reload();
					IMessageBox.toast("销户成功！");
				});
			}
		});
    },
    setDepartment:function(){//部门设置
    	
    	var row = this.getSelectedItem();
    	if(row == null){
    		
    		IMessageBox.warning("请选择要操作的记录！");
    		return;
    	}
    	if(row.status == '未开户'){
    		
    		IMessageBox.warning("服务商未开户！");
    		return;
    	}
    	if(row.status == '已注销'){
    		
    		IMessageBox.warning("服务商已注销！");
    		return;
    	}
    	window.open('/panda/operation/supplier/department/list?supplierId='+row.id);
    },
    setSalesman:function(){//员工设置

    	var row = this.getSelectedItem();
    	if(row == null){
    		
    		IMessageBox.warning("请选择要操作的记录！");
    		return;
    	}
    	if(row.status == '未开户'){
    		
    		IMessageBox.warning("服务商未开户！");
    		return;
    	}
    	if(row.status == '已注销'){
    		
    		IMessageBox.warning("服务商已注销！");
    		return;
    	}
    	window.open('/panda/operation/supplier/salesman/list?supplierId='+row.id);
    }
});